### PR TITLE
Add exception handler hook

### PR DESCRIPTION
**Pull request by mjeffrey18**
Allow customisation of exception handling

Also mentioned here https://github.com/cable-cr/cable/issues/47

### CODE REVIEW

**Comment by jwoertink** [comment]
Those are some interesting errors in Crystal 1.0 and 1.1... :joy:  I wonder if maybe we just should make the next release require Crystal 1.4+ We did that for Lucky following Ameba's lead.

**Review by jwoertink — APPROVED** [review]

**Comment by fernandes** [comment]
> require Crystal 1.4+

I'm 👍

**Review by fernandes — APPROVED** [review]

File: src/backend/redis/legacy/backend.cr [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # On redis shard it tries to convert the return of command to Nil
 # When returning an array, it raises an exception
 # So we monkey patch to run the command, ignore it, and return Nil
+{% if Redis.class? %}
```
Comment (jwoertink) [inline src/backend/redis/legacy/backend.cr:4]: ooo clever!

**Review by jwoertink — APPROVED** [review]
Nice :+1:  This will be awesome to have in

**Pull request merged**